### PR TITLE
Fixes for luaL_ref (resolves #80)

### LINKED ISSUE #80: luaL_ref(L, lua.LUA_REGISTRYINDEX) fails
Hi, I'm trying to use the Lua's reference system as described in https://www.lua.org/pil/27.3.2.html, but luaL_ref throws an assertion error:
```
AssertionError: false == true
    at Object.luaH_getint (webpack-internal:///11:153:5)
    at Object.lua_rawgeti (webpack-internal:///63:478:33)
    at Object.luaL_ref (webpack-internal:///5:626:13)
```

Chrome's debugger screenshot:
image

Here's a simplified test case:

```javascript
"use strict";

const test       = require('tape');

global.WEB = false;

const lauxlib    = require("../../src/lauxlib.js");
const lua        = require('../../src/lua.js');

test('luaL_ref, lua_rawgeti, luaL_unref, LUA_REGISTRYINDEX', function( t ) {
    let L;

    t.plan(2)

    t.doesNotThrow(function () {
        L = lauxlib.luaL_newstate();
        lua.lua_pushstring(L, lua.to_luastring("hello references!"))

        let r = lauxlib.luaL_ref(L, lua.LUA_REGISTRYINDEX) // pops a value, stores it and returns a reference
        lua.lua_rawgeti(L, lua.LUA_REGISTRYINDEX, r) // pushes a value associated with the reference
        lauxlib.luaL_unref(L, lua.LUA_REGISTRYINDEX, r) // releases the reference

    }, "JS Lua program ran without error")

    t.strictEqual(
        lua.lua_tojsstring(L, -1),
        "hello references!",
        "top is correct"
    )
});
```

### PR DESCRIPTION
**Pull request by dlbd**
Fixes #80

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,10 @@ In the browser `require` and `package.loadlib` try to find a file by making sync
     - `os.execute`
 - [Weak tables](http://www.lua.org/manual/5.3/manual.html#2.5.2)
```
Comment (daurnimator) [inline README.md]: Could you add a 2nd blank line here? There should be two blank lines at the end of a section. (also need a 2nd empty line after your addition)

File: src/lapi.js [hunk 2]
```diff
@@ -708,6 +708,11 @@ const lua_tocfunction = function(L, idx) {
 };
 
 const lua_tointeger = function(L, idx) {
+    let n = lvm.tointeger(index2addr(L, idx));
```
Comment (daurnimator) [inline src/lapi.js]: Perhaps this should call `lua_tointegerx`?

**Comment by daurnimator** [comment]
Could you edit your commit messages to be prefixed with the main file(s) changed?

File: src/lapi.js (hunk 2 above)
Comment (dlbd) [reply src/lapi.js]: Perhaps, C Lua does ``#define lua_tointeger(L,i)      lua_tointegerx(L,(i),NULL)`` and ``
#define lua_tonumber(L,i)       lua_tonumberx(L,(i),NULL)``

**Comment by dlbd** [comment]
Pull request updated.

**Comment by daurnimator** [comment]
Thanks for your contribution!

**Pull request merged**